bot/exts/events/advent_of_code/_cog.py

```diff
@@ -52,13 +60,80 @@ def __init__(self, bot: Bot):
+        Runs on a schedule, as defined in the task.loop decorator.
+        """
+        await self.bot.wait_until_guild_available()
+        guild = self.bot.get_guild(Client.guild)
+        completionist_role = guild.get_role(Roles.aoc_completionist)
+
+        aoc_name_to_member_id = {
+            aoc_name: member_id
+            for member_id, aoc_name in await self.account_links.items()
+        }
+
+        try:
+            leaderboard = await _helpers.fetch_leaderboard()
+        except _helpers.FetchingLeaderboardFailedError:
+            await self.bot.send_log("Unable to fetch AoC leaderboard during role sync.")
+            return
+
+        placement_leaderboard = json.loads(leaderboard["placement_leaderboard"])
+        members_who_deserve_role: set[discord.Member] = set()
+
+        for member_aoc_info in placement_leaderboard.values():
+            if not member_aoc_info["stars"] == 50:
+                # Only give the role to people who have completed all 50 stars
+                continue
+
+            member_id = aoc_name_to_member_id.get(member_aoc_info["name"], None)
+            if not member_id:
+                continue
+
+            member = await members.get_or_fetch_member(guild, member_id)
+            if member is None:
+                continue
+
+            if member in completionist_role.members:
+                if await self.completionist_block_list.contains(member_id):
+                    await members.handle_role_change(member, member.remove_roles, completionist_role)
+                continue
+
+            await members.handle_role_change(member, member.add_roles, completionist_role)
+            members_who_deserve_role.add(member)
+
+        for member in set(completionist_role.members) - members_who_deserve_role:
+            # Remove the role from members who have it but don't deserve it (been removed from the leaderboard)
+            await members.handle_role_change(member, member.remove_roles, completionist_role)
+
     @commands.group(name="adventofcode", aliases=("aoc",))
     @whitelist_override(channels=AOC_WHITELIST)
     async def adventofcode_group(self, ctx: commands.Context) -> None:
         """All of the Advent of Code commands."""
         if not ctx.invoked_subcommand:
             await invoke_help_command(ctx)
 
+    @with_role(Roles.admins)
+    @adventofcode_group.command(
+        name="block",
+        brief="Block a user from getting the completer role.",
+    )
+    async def block_from_role(self, ctx: commands.Context, member: discord.Member) -> None:
+        """Block the given member from receiving the AoC completionist role, removing it from them if needed."""
+        await self.bot.wait_until_guild_available()
+        completionist_role = ctx.guild.get_role(Roles.aoc_completionist)
+        if completionist_role in member.roles:
+            await member.remove_roles(completionist_role)
+
+        await self.completionist_block_list.set(member.id, None)
```

Comment:
```
sir-lancebot          | 2021-12-28 20:14:11 | root | DEBUG | Error Encountered: TypeError - RedisObject._to_typestring only supports the following: (('f|', <class 'float'>), ('i|', <class 'int'>), ('s|', <class 'str'>), ('b|', <class 'bool'>))., Command: adventofcode block, Author: Mark#3118, Channel: general-meta-voice
sir-lancebot          | 2021-12-28 20:14:11 | bot.exts.core.error_handler | ERROR | Unhandled command error: RedisObject._to_typestring only supports the following: (('f|', <class 'float'>), ('i|', <class 'int'>), ('s|', <class 'str'>), ('b|', <class 'bool'>)).
sir-lancebot          | Traceback (most recent call last):
sir-lancebot          |   File "/usr/local/lib/python3.9/site-packages/discord/ext/commands/core.py", line 167, in wrapped
sir-lancebot          |     ret = await coro(*args, **kwargs)
sir-lancebot          |   File "/bot/bot/exts/events/advent_of_code/_cog.py", line 135, in block_from_role
sir-lancebot          |     await self.completionist_block_list.set(member.id, None)
sir-lancebot          |   File "/usr/local/lib/python3.9/site-packages/async_rediscache/types/base.py", line 260, in wrapper
sir-lancebot          |     result = await coroutine_object
sir-lancebot          |   File "/usr/local/lib/python3.9/site-packages/async_rediscache/types/cache.py", line 75, in set
sir-lancebot          |     value = self._value_to_typestring(value)
sir-lancebot          |   File "/usr/local/lib/python3.9/site-packages/async_rediscache/types/base.py", line 128, in _to_typestring
sir-lancebot          |     raise TypeError(f"RedisObject._to_typestring only supports the following: {prefixes}.")
sir-lancebot          | TypeError: RedisObject._to_typestring only supports the following: (('f|', <class 'float'>), ('i|', <class 'int'>), ('s|', <class 'str'>), ('b|', <class 'bool'>)).
```

Reply:
Fixed in https://github.com/python-discord/sir-lancebot/pull/991/commits/2fe89228fd3e583f2c8746d7df08af9c1c234972